lib/rules/function-calc-no-unspaced-operator/index.cjs

```diff
@@ -65,169 +62,301 @@ const rule = (primary, _secondaryOptions, context) => {
-			 * @param {number} index
-			 * @param {'before' | 'after'} position
-			 * @returns {void}
-			 */
-			function checkWhitespaceAroundOperator(operatorNode, parent, index, position) {
-				const isBefore = position === 'before';
-				const aroundNode = parent.value[index + (isBefore ? -1 : 1)];
+			const tokens = cssTokenizer.tokenize({ css: value });
 
-				if (cssParserAlgorithms.isWhitespaceNode(aroundNode)) {
-					const whitespace = getWhitespace(aroundNode);
+			{
+				// Step 1
+				// Step 1.1
+				// Re-tokenize dimensions with units containing dashes.
+				// These might be typo's.
+				// For example: `10px-20px` has a unit of `px-20px`
+				for (let i = 0; i < tokens.length; i++) {
+					const token = tokens[i];
 
-					if (whitespace === ' ') return;
+					// NOTE:
+					// This is a mess, I am fairly certain that it is correct, but too much is happening with strings.
+					// This will be brittle and might break when the css-tokenizer is updated.
+					// It will also produce invalid CSS for exotic units, nu such units exists today.
+					if (!token || token[0] !== cssTokenizer.TokenType.Dimension) continue;
 
-					const indexOfFirstNewLine = whitespace.search(NEWLINE_REGEX);
+					if (token[4].unit.startsWith('--')) continue;
 
-					if (indexOfFirstNewLine === 0) return;
+					const indexOfDash = token[4].unit.indexOf('-');
 
-					if (context.fix) {
-						needsFix = true;
+					if (indexOfDash === -1) continue;
 
-						const [token] = aroundNode.value;
+					const remainder = token[4].unit.slice(indexOfDash);
 
-						if (token) {
-							token[1] = indexOfFirstNewLine === -1 ? ' ' : whitespace.slice(indexOfFirstNewLine);
-						}
+					if (remainder.length === 1) continue;
 
-						return;
-					}
-				} else if (
-					cssParserAlgorithms.isTokenNode(aroundNode) &&
-					(isOperandNode(aroundNode) || isScssInterpolation(aroundNode, parent))
-				) {
-					if (context.fix) {
-						needsFix = true;
+					token[4].unit = token[4].unit.slice(0, indexOfDash);
+					token[1] = `${token[4].signCharacter === '+' ? '+' : ''}${token[4].value}${
+						token[4].unit
+					}`;
+					token[3] = token[2] + token[1].length;
 
-						if (isBefore) {
-							aroundNode.value[1] += ' ';
-						} else {
-							aroundNode.value[1] = ` ${aroundNode.value[1]}`;
-						}
+					const remainderTokens = cssTokenizer.tokenize({ css: remainder }).slice(0, -1); // Trim EOF token
 
-						return;
-					}
-				} else {
-					return;
-				}
+					remainderTokens.forEach((remainderToken) => {
+						remainderToken[2] += token[3];
+						remainderToken[3] += token[3];
+					});
 
-				const delimToken = getDelimToken(operatorNode);
+					tokens.splice(i + 1, 0, ...remainderTokens);
+				}
 
-				if (!delimToken) return;
+				// Step 1.2
+				// Re-tokenize scss interpolation blocks
+				// Grouping `#` and `{` into a single token allows us to parse these as simple blocks with curly braces.
```

Comment:
question

Would less be supported just by doing the same thing for `@`?
e.g. `@{variable}`

Reply:
> So anything that can easily become such a simple block is also easy to ignore.

If someone ever request it, we will.

Reply:
> But I am unsure if we want to increase our exposure to non-standard syntaxes.

For your reference, we want to support *only* standard syntax. Supporting non-standard syntaxes should be exceptional, e.g. for backward compatibility.

See also the [document](https://stylelint.io/developer-guide/rules#define-the-rule):

> for standard CSS syntax only